waggle-dance-integration-tests/src/test/java/com/hotels/bdp/waggledance/WaggleDanceIntegrationTest.java

```diff
@@ -272,6 +274,42 @@ public void usePrefix() throws Exception {
     assertTypicalRemoteTable(proxy, waggledRemoteDbName);
   }
 
+  @Test
+  public void manyFederatedMetastores() throws Exception {
+    runner = WaggleDanceRunner
+        .builder(configLocation)
+        .databaseResolution(DatabaseResolution.PREFIXED)
+        .primary("primary", localServer.getThriftConnectionUri(), READ_ONLY)
+        .federate(SECONDARY_METASTORE_NAME, remoteServer.getThriftConnectionUri())
+        .federate("fed1", remoteServer.getThriftConnectionUri())
+        .federate("fed2", remoteServer.getThriftConnectionUri())
+        .federate("fed3", remoteServer.getThriftConnectionUri())
+        .federate("fed4", remoteServer.getThriftConnectionUri())
+        .federate("fed5", remoteServer.getThriftConnectionUri())
+        .federate("fed6", remoteServer.getThriftConnectionUri())
+        .federate("fed7", remoteServer.getThriftConnectionUri())
+        .federate("fed8", remoteServer.getThriftConnectionUri())
+        .federate("fed9", remoteServer.getThriftConnectionUri())
+        .federate("fed10", remoteServer.getThriftConnectionUri())
+        .federate("fed11", remoteServer.getThriftConnectionUri())
+        .federate("fed12", remoteServer.getThriftConnectionUri())
+        .federate("fed13", remoteServer.getThriftConnectionUri())
```

Comment:
Is any of these metastores failing or taking more than expected? So we could test also the part of waiting 1 minute and logging?

Reply:
No but great question,
That's not that easy to do, I'll have to think about that. Added this for now to cover ordering and easy debugging.
Might need to add something as part of a QA suite, or create a very slow metastore server in the unit tests (which we can't do at this moment).